src/controllers/rpc/publish-rpc-controller.js

```diff
@@ -27,21 +27,28 @@ class PublishController extends BaseController {
 
                 break;
             case NETWORK_MESSAGE_TYPES.REQUESTS.PROTOCOL_REQUEST:
-                // eslint-disable-next-line no-case-declarations
-                dataSource = await this.operationIdService.getCachedOperationIdData(operationId);
-                await this.operationIdService.cacheOperationIdData(operationId, {
-                    assertionId: dataSource.assertionId,
-                    assertion: message.data.assertion,
-                });
-                command.name = handleRequestCommand;
-                command.data.keyword = message.data.keyword;
-                command.data.agreementId = dataSource.agreementId;
-                command.data.agreementData = dataSource.agreementData;
+                try {
+                    // eslint-disable-next-line no-case-declarations
+                    dataSource = await this.operationIdService.getCachedOperationIdData(
+                        operationId,
+                    );
+                    await this.operationIdService.cacheOperationIdData(operationId, {
+                        assertionId: dataSource.assertionId,
+                        assertion: message.data.assertion,
+                    });
+                    command.name = handleRequestCommand;
+                    command.data.keyword = message.data.keyword;
+                    command.data.agreementId = dataSource.agreementId;
+                    command.data.agreementData = dataSource.agreementData;
+                } catch (error) {
```

Comment:
The error should somehow be handled aswell, otherwise the handler command will be scheduled with invalid data in the cache.